scripts/ci/check_compliance.py

```diff
@@ -1447,6 +1447,64 @@ def run(self):
             with open(file, "r") as fp:
                 self.check_file(file, fp)
 
+class FileEncoding(ComplianceTest):
+    """
+    Check the encoding for source files.
+    """
+    name = "FileEncoding"
+    doc = "Check the encoding for source files."
+    path_hint = "<git-top>"
+
+    MATCH_FILES = [
+            r'.*\.c$',
+            r'.*\.h$',
+            r'.*\.sh$',
+            r'.*\.yaml$',
+            r'.*Kconfig.*',
+            r'.*CMakeLists.txt$',
+            ]
+
+    def match_file(self, file):
+        for match in self.MATCH_FILES:
+            if re.match(match, file):
+                return True
+
+        return False
+
+    def check_file(self, file, fp):
+        if not self.match_file(str(file)):
+            return
+
+        for line_num, line in enumerate(fp.readlines(), start=1):
+            line = line.strip()
+
+            if not line:
+                continue
+
+            if "Copyright" in line:
+                continue
+
+            try:
+                line.encode().decode(encoding="ascii", errors="strict")
```

Comment:
Consider using `line.isascii()` instead of the encode-decode dance. Throw in `and line.isprintable()` for good measure.

Reply:
Hey thanks for the tip, can't believe that function was there the whole time, changed.